sql-plugin/src/main/scala/com/nvidia/spark/rapids/CostBasedOptimizer.scala

```diff
@@ -211,89 +234,109 @@ trait CostModel {
    * @param plan Operator
    * @return (cpuCost, gpuCost)
    */
-  def applyCost(plan: SparkPlanMeta[_]): (Double, Double)
-
-  /**
-   * Determine the cost of transitioning data from CPU to GPU for a specific operator
-   * @param plan Operator
-   * @return Cost
-   */
-  def transitionToGpuCost(plan: SparkPlanMeta[_]): Double
+  def getCost(plan: SparkPlanMeta[_]): Double
 
-  /**
-   * Determine the cost of transitioning data from GPU to CPU for a specific operator
-   */
-  def transitionToCpuCost(plan: SparkPlanMeta[_]): Double
 }
 
-class DefaultCostModel(conf: RapidsConf) extends CostModel {
+class CpuCostModel(conf: RapidsConf) extends CostModel {
 
-  def transitionToGpuCost(plan: SparkPlanMeta[_]) = {
-    // this is a placeholder for now - we would want to try and calculate the transition cost
-    // based on the data types and size (if known)
-    conf.defaultTransitionToGpuCost
+  private val attrRefCost = conf.getCpuExpressionCost("AttributeReference").getOrElse(0d)
+  private val structFieldCost = conf.getCpuExpressionCost("GetStructField").getOrElse(0.05d)
```

Comment:
I'm not crazy about these different levels of configs but not sure what the right answer is. I don't think we want to add a specific config for every single operator and expression so having a generic mechanism seems to make sense for now but then we need defaults as well.

Reply:
Having configs for each of these things is really only going to help with testing. No real user will be able to set these in any meaningful way, not should they.